### PR TITLE
Unify deser impl

### PR DESCRIPTION
**Pull request by ijackson**
As part of my efforts to try to make `#[deser(flatten)]` work correctly (#336) I did some refactoring.  Although the deser work is unfinished, I think these two commits are worth submitting separately.

### CODE REVIEW

**Review by matthiasbeyer — CHANGES_REQUESTED** [review]
Hi. Sorry for not getting back to this for such a long time... I had other things on my mind and then I was on vacation. :beach_umbrella: 

I hope you don't mind. :smile_cat: 

If you are not interested anymore in continuing work on this PR, I am happy to make the adaptions requested in my review myself!

File: tests/errors.rs [hunk 1]
```diff
@@ -31,6 +31,35 @@ fn test_error_parse() {
...
+            res.unwrap_err().to_string(),
+            format!(
+                "invalid type: string \"Torre di Pisa\", expected an integer for key `place.name` in tests/Settings.toml",
+            )
+        );
+    };
+
+    let c = make();
+    let res = c.try_deserialize().map(|_: Output| ());
+    expect_err(res);
```
[the remark this answers is not in the document]
Comment (ijackson) [reply tests/errors.rs:60]: I was expecting there might be more tests with errors, so wanted to provide a helper to check an error was expected.  I can inline this again if you prefer, which would probably make it a bit simpler looking.

File: src/de.rs [hunk 2]
```diff
@@ -337,132 +337,91 @@ impl<'de> de::VariantAccess<'de> for EnumAccess {
...
-
-    #[inline]
-    fn deserialize_u32<V: de::Visitor<'de>>(self, visitor: V) -> Result<V::Value> {
-        // FIXME: This should *fail* if the value does not fit in the requets integer type
-        visitor.visit_u32(self.cache.into_int()? as u32)
-    }
-
-    #[inline]
-    fn deserialize_u64<V: de::Visitor<'de>>(self, visitor: V) -> Result<V::Value> {
-        // FIXME: This should *fail* if the value does not fit in the requets integer type
```
[the remark this answers is not in the document]
Comment (ijackson) [reply src/de.rs:411]: In the meantime those FIXMEs have gone and much of the code has been reorganised.  I'm going to fix #464 first and then maybe revisit this.

File: tests/errors.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (matthiasbeyer) [reply tests/errors.rs:60]: Yes do that please.

**Comment by ijackson** [comment]
Closing in favour of #488

**Pull request closed without merging**